src/util/plotting.jl

```diff
@@ -115,18 +118,16 @@ SamplePlot((f,)::Tuple{<:FiniteGP}) = SamplePlot((f.x, f))
 SamplePlot((x, gp)::Tuple{<:AbstractVector,<:AbstractGP}) = SamplePlot((gp(x, 1e-9),))
 
 @recipe function f(sp::SamplePlot)
-    nsamples::Int = get(plotattributes, :samples, 1)
+    nsamples::Int = pop!(plotattributes, :samples, 1)
     samples = rand(sp.f, nsamples)
 
+    flat_x = vcat(Iterators.flatten(zip(Iterators.repeated(sp.x), fill(NaN, nsamples)))...)
+    flat_f = vcat(Iterators.flatten(zip(eachcol(samples), Iterators.repeated([NaN])))...)
```

Comment:
@devmotion happy for you to suggest a better way - this is what I could come up with so far ;)

Reply:
Apparently the splatting is bad? but I can't figure out a way to get around it

Reply:
My suggestion would be
```suggestion
    flat_x = repeat(vcat(sp.x, NaN), nsamples)
    flat_f = vec(vcat(samples, fill(NaN, 1, nsamples)))
```

Reply:
ah, I literally just came upon the same solution myself right before looking at your comment :tada:

Reply:
Except the "interleave with NaN" doesn't actually seem to work, as `seriestype = :line` makes it sort by x :(

So we could remove the `seriestype --> :line` below, but then you must pass in an ordered `x` for the `sampleplot` to make any sense. Would you be happy with _that_?

Reply:
We could also just sort `x`, couldn't we?